src/lib/web-component/filter-keyword/filter-keyword-element.js

```diff
@@ -0,0 +1,39 @@
+export class FilterKeyword extends HTMLElement {
+  constructor() {
+    super();
+
+    const shadow = this.attachShadow({ mode: 'open' });
+    const filterContainer = document
+      .querySelector('#filterKeywordTemplate')
+      .content.cloneNode(true);
+
+    this.inputBox = filterContainer.querySelector('input[name="keyword"]');
+
+    this.timer = 0;
+    this.inputBoxListener = (event) => {
+      clearTimeout(this.timer);
+      this.timer = setTimeout(() => {
+        const filterDetail = {
+          updateFilter: { keyword: event.target.inputBox.value },
+        };
+
+        this.dispatchEvent(
+          new CustomEvent('filterchange', { detail: filterDetail })
+        );
```

Comment:
Move the logic to a `dispatchFilterChanged` method. You don't need to use `event.target.inputBox`, `this.inputBox` suffices.